### DISCUSSION TITLE
How to provide globalPassThroughEnv at the extended package level?

### DISCUSSION THREAD

**Discussion by gajus** [vercel/turborepo#6482] [Help] [answered]
Effectively, I want to make `CDA_*` available only to the current workspace.

Currently, I do this by enumerating every task:

```json
{
  "$schema": "https://turbo.build/schema.json",
  "extends": [
    "//"
  ],
  "pipeline": {
    "build": {
      "env": [
        "CDA_*"
      ]
    },
    "dev": {
      "env": [
        "CDA_*"
      ]
    },
    "generate": {
      "env": [
        "CDA_*"
      ]
    },
    "test:vitest": {
      "env": [
        "CDA_*"
      ]
    }
  }
}
```

Surely there is a better way?

**Comment by mehulkar** [comment, marked as answer]
There is currently no better way. When we (I) did workspace configs, I ran into a ton of design problems with enabling global keys in the workspace config. It felt odd to allow _some_ global keys, but not all (e.g. what does it mean to have `remoteCache` keys in a specific workspace?). So we (I) scoped the work down to only allow task configs.

That said, this is a reasonable ask. The next step is to add a feature to enable the global keys that make sense in Workspace Configs. Feel free to file a Feature request for that! I imagine for your use case, we'd want to enable both `globalDotEnv` and `globalEnv`, but I feel like we may need to remove the `global` prefixes? 

(Since this started as a discussion, I'll close it out since the answer is "no it's not possible today", and we can continue in the feature request if you file it!)
